shell/platform/android/io/flutter/view/TextureRegistry.java

```diff
@@ -57,7 +66,45 @@ interface TextureEntry {
     void release();
   }
 
+  interface ImageProducerTextureEntry extends TextureEntry {
+    /**
+     * Retrieve the last Image produced. Drops all previously produced images.
+     *
+     * <p>NOTE: Caller must call Image.close() on returned image.
+     *
+     * @return Image or null.
+     */
+    public Image acquireLatestImage();
+  }
+
+  /** A registry entry for texture updated by rendering into a Surface. */
+  @Keep
+  interface RawSurfaceTextureEntry extends TextureEntry {
+    /** @return The identity of this texture. */
+    long id();
```

Comment:
Texture or surface? or just "return the identity of this resource/entry"

Reply:
Texture. This is the id that we use for linkage with the framework. Not related to the surface at all.